File: src/main/java/me/dyzjct/kura/module/modules/combat/EzBow.java
package me.dyzjct.kura.module.modules.combat;

import me.dyzjct.kura.event.events.client.PacketEvents;
import me.dyzjct.kura.module.Category;
import me.dyzjct.kura.module.Module;
import me.dyzjct.kura.setting.BooleanSetting;
import me.dyzjct.kura.setting.Setting;
import net.minecraft.item.ItemBow;
import net.minecraft.item.ItemStack;
import net.minecraft.network.play.client.CPacketEntityAction;
import net.minecraft.network.play.client.CPacketPlayer;
import net.minecraft.network.play.client.CPacketPlayerDigging;
import net.minecraft.network.play.client.CPacketPlayerTryUseItem;
import net.minecraft.util.EnumHand;
import net.minecraft.util.math.BlockPos;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

@Module.Info(name = "EzBow", category = Category.COMBAT)
public class EzBow extends Module {
    public BooleanSetting bow = bsetting("NoDelay", false);
    public Setting<Integer> spoofs = isetting("Spoofs", 10, 1, 300);

    @Override
    public void onUpdate() {
        if (fullNullCheck()) return;
        if (bow.getValue() && mc.player.getHeldItemMainhand().getItem() instanceof ItemBow && mc.player.isHandActive() && mc.player.getItemInUseMaxCount() >= 3) {
            mc.player.connection.sendPacket(new CPacketPlayerDigging(CPacketPlayerDigging.Action.RELEASE_USE_ITEM, BlockPos.ORIGIN, mc.player.getHorizontalFacing()));
            mc.player.connection.sendPacket(new CPacketPlayerTryUseItem(mc.player.getActiveHand()));
            mc.player.stopActiveHand();
        }
        if (bow.getValue() && mc.player.getHeldItemOffhand().getItem() instanceof ItemBow && mc.player.isHandActive() && mc.player.getItemInUseMaxCount() >= 3) {
            mc.player.connection.sendPacket(new CPacketPlayerDigging(CPacketPlayerDigging.Action.RELEASE_USE_ITEM, BlockPos.ORIGIN, mc.player.getHorizontalFacing()));
            mc.player.connection.sendPacket(new CPacketPlayerTryUseItem(mc.player.getActiveHand()));
            mc.player.stopActiveHand();
        }
    }

    @SubscribeEvent
    public void onPacketSend(PacketEvents.Send event) {
        if (event.getStage() != 0) {
            return;
        }
        if (event.getPacket() instanceof CPacketPlayerDigging) {
            CPacketPlayerDigging packet = event.getPacket();
            if (packet.getAction() == CPacketPlayerDigging.Action.RELEASE_USE_ITEM) {
                ItemStack handStack = mc.player.getHeldItem(EnumHand.MAIN_HAND);
                if (!handStack.isEmpty() && handStack.getItem() != null && handStack.getItem() instanceof ItemBow) {
                    mc.player.connection.sendPacket(new CPacketEntityAction(mc.player, CPacketEntityAction.Action.START_SPRINTING));
                    for (int index = 0; index < spoofs.getValue(); ++index) {
                        mc.player.connection.sendPacket(new CPacketPlayer.PositionRotation(mc.player.posX, mc.player.posY + 1e-10, mc.player.posZ, mc.player.rotationYaw, mc.player.rotationPitch, false));
                        mc.player.connection.sendPacket(new CPacketPlayer.PositionRotation(mc.player.posX, mc.player.posY - 1e-10, mc.player.posZ, mc.player.rotationYaw, mc.player.rotationPitch, true));
                    }
                }
            }
        }
    }

}

File: src/main/java/me/windyteam/kura/module/modules/combat/EzBow.kt
package me.windyteam.kura.module.modules.combat

import me.windyteam.kura.event.events.client.PacketEvents
import me.windyteam.kura.module.Category
import me.windyteam.kura.module.Module
import me.windyteam.kura.utils.mc.ChatUtil
import net.minecraft.entity.Entity
import net.minecraft.item.ItemBow
import net.minecraft.item.ItemStack
import net.minecraft.network.Packet
import net.minecraft.network.play.client.CPacketEntityAction
import net.minecraft.network.play.client.CPacketPlayer.PositionRotation
import net.minecraft.network.play.client.CPacketPlayerDigging
import net.minecraft.util.EnumHand
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent

@Module.Info(name = "32kBow", category = Category.COMBAT)
object EzBow : Module() {
    private var spoofs = settings("Spoofs", 10, 1, 300)
    private var debug = settings("Debug", false)
    var cancel = false

    @SubscribeEvent
    fun onPacketSend(event: PacketEvents.Send) {
        if (event.stage !== 0) {
            return
        }
        if (event.packet is CPacketPlayerDigging) {
            val packet = event.getPacket() as CPacketPlayerDigging
            if (packet.action == CPacketPlayerDigging.Action.RELEASE_USE_ITEM) {
                val handStack: ItemStack = mc.player.getHeldItem(EnumHand.MAIN_HAND)
                if (!handStack.isEmpty() && handStack.getItem() != null && handStack.getItem() is ItemBow) {
                    cancel = true
                    mc.player.connection.sendPacket(
                        CPacketEntityAction(
                            mc.player as Entity,
                            CPacketEntityAction.Action.START_SPRINTING
                        ) as Packet<*>
                    )
                    for (index in 0 until spoofs.value) {
                        mc.player.connection.sendPacket(
                            PositionRotation(
                                mc.player.posX,
                                mc.player.posY + 1.0E-10,
                                mc.player.posZ,
                                mc.player.rotationYaw,
                                mc.player.rotationPitch,
                                false
                            ) as Packet<*>
                        )
                        mc.player.connection.sendPacket(
                            PositionRotation(
                                mc.player.posX,
                                mc.player.posY - 1.0E-10,
                                mc.player.posZ,
                                mc.player.rotationYaw,
                                mc.player.rotationPitch,
                                true
                            ) as Packet<*>
                        )
                        if (debug.value) ChatUtil.sendMessage("Spoofed")
                    }
                }
            } else {
                cancel = false
            }
        } else {
            cancel = false
        }
    }
}